### ISSUE TITLE
Add a minimum dimmer setting to standard/genaric dimmer DTHs

### ISSUE THREAD

**Issue by vseven** [SmartThingsCommunity/SmartThingsPublic#2406]
With non-smart dimmers, such as a Lutron, Leviton, etc, the hardwired dimmers usually have a minimum adjustment knob to reduce flicker.  However most "smart" dimmers don't have this ability and depending on the manufacture there is no parameter to change.  What that means is potentially the dimmer can be set below a bulbs minimum causing flickering or sometimes the bulb simply not working.

I've recently had this issue with the GE/JASCO Z-Wave Plus dimmer.  The non-plus version worked fine with my Phillips Warm Glow candelabra bulbs but the new Plus version has bad flicker below 10%.  After contacting GE/JASCO they told me there was no parameter in the switch that could be changed and that the minimum should be enforced from the Hub.   I realize this isn't a perfect answer because there would be flicker even without a smart hub (a.k.a SmartThings) present but I've had the same issue with other dimmer and other bulbs.   So the ability to force minimums would be nice.

I'm proposing a additional preference to the standard DTH's used for dimming that can be set to a minimum value.  I.e. a preference can be set to say 15% so the dimmer cannot go below 15%.   If its sent lower it simply sends the higher.  For example in the standard "Dimmer Switch" DTH this is the current setLevel command:

```
def setLevel(value) {
	log.debug "setLevel >> value: $value"
	def valueaux = value as Integer
	def level = Math.max(Math.min(valueaux, 99), 0)
	if (level > 0) {
		sendEvent(name: "switch", value: "on")
	} else {
		sendEvent(name: "switch", value: "off")
	}
	sendEvent(name: "level", value: level, unit: "%")
	delayBetween ([zwave.basicV1.basicSet(value: level).format(), zwave.switchMultilevelV1.switchMultilevelGet().format()], 5000)
}
```

I'm proposing a change to:

```
def setLevel(value) {
	log.debug "setLevel >> value: $value"
	def valueaux = value as Integer
	def level = Math.max(Math.min(valueaux, 99), 0)
	if (level > 0) {
                if (level < $minLevelPref) {
                          level = $minLevelPref
                }
		sendEvent(name: "switch", value: "on")
	} else {
		sendEvent(name: "switch", value: "off")
	}
	sendEvent(name: "level", value: level, unit: "%")
	delayBetween ([zwave.basicV1.basicSet(value: level).format(), zwave.switchMultilevelV1.switchMultilevelGet().format()], 5000)
}
```

In the above if the level being set is above 0 it then does a second check to see if the value is below the minimum.  If it is then the value gets set to the minimum.  This would be a very simple change but would increase usability and I don't see a downside as it would be a user preference that would default to 0.  Only other addition I can see is adding a == Null and if so making it 0.

**Comment by workingmonk** [comment]
i agree with the problem, but not sure how the preferences would be handled when running locally. tagging @tpmanley and @larsfinander for comment, but I believe that this may be the point where we say "this is why community DTHs exist and can override default ones to support specific use cases".

**Comment by vseven** [comment]
Does the temperature offset on the Temp/Humidity sensors offset locally?  If so wouldn't it be the same thing?

I realize you can put it into a custom DTH but the whole reason would be to maintain "standard" DTH's that run locally.  It solves a problem that I know people have had and I can't see how it would create any issue, although thats why I I'm asking here.  :)
